errormessages.grace

```diff
@@ -440,3 +440,29 @@ method syntaxError(message)atLine(errlinenum)withSuggestions(suggestions) {
     }
     util.syntaxError(message, errlinenum, "", arr, false, suggestions)
 }
+
+method canonicalMethodName(name) {
```

Comment:
This will work, and the algorithm is fine.  But in general it's a bad idea to index into strings.  Remember that they are Unicode strings, and characters are of a variable number of bytes.  So each indexing operation requires scanning the string from the start.  

It's better to use an iterator:
```grace    
def iter = name.iterator
var ch
while { iter.hasNext } do {
    ch := iter.next
    ...
}
```
Th lines incrementing `i` will go away, and I think that you will find the code shorter and clearer as well as faster.